csharp/src/Apache.Arrow/Memory/LongMemoryOwner.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Apache.Arrow.Memory
+{
+    internal class LongMemoryOwner<T>: IMemoryOwner<T>
```

Comment:
I don't think we want to define a LongMemoryOwner class, but rather an `ILongMemoryOwner` interface similar to `IMemoryOwner`